core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -0,0 +1,100 @@
+package io.grpc;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * TODO: Insert description here. (generated by notcarl)
+ */
+public final class MessageEncoding {
+
+  /**
+   * Special sentinel codec indicating that no compression should be used.  Users should use
+   * reference equality to see if compression is disabled.
+   */
+  public static final Codec NONE = new Codec() {
+    @Override
+    public InputStream decompress(InputStream is) throws IOException {
+      return is;
+    }
+
+    @Override
+    public String getMessageEncoding() {
+      return null;
+    }
+
+    @Override
+    public OutputStream compress(OutputStream os) throws IOException {
+      return os;
+    }
+  };
+
+
+  /**
+   * Represents a message compressor.
+   */
+  public interface Compressor {
```

Comment:
@carl-mastrangelo 

> Using such a method would be specific to compressing, but not decompressing, ...

If that's the case, then I would expect `Codec` to use composition:

```java
interface Codec {
    String name();
    Compressor compressor();
    Decompressor decompressor();
}

interface Compressor {
    OutputStream compress(OutputStream os) throws IOException;
}

interface Decompressor {
    InputStream decompress(InputStream is) throws IOException;
}
```